File: todo/test_gmt_ifft_2d.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <complex.h>
#include "gmt.h"

int main() {
    void *API;
    unsigned int n_columns = 4;
    unsigned int n_rows = 4;
    int direction = GMT_FFT_INV; // Inverse transform
    float complex data[] = {136 + 0*I, -8 + 8*I, -8 + 0*I, -8 - 8*I,
                            -32 + 32*I, 0 + 0*I, 0 + 0*I, 0 + 0*I,
                            -32 + 0*I, 0 + 0*I, 0 + 0*I, 0 + 0*I,
                            -32 - 32*I, 0 + 0*I, 0 + 0*I, 0 + 0*I};

    // Initialize the GMT API
    if ((API = GMT_Create_Session("test_gmt_fft_2d", 2, 0, NULL)) == NULL) {
        printf("Error: Could not initialize the GMT API.\n");
        return EXIT_FAILURE;
    }

    // Call the GMT_FFT_2D function
    if (GMT_FFT_2D(API, (float *)data, n_columns, n_rows, direction, GMT_FFT_COMPLEX) != GMT_NOERROR) {
        printf("Error: Could not perform the 2D FFT.\n");
        return EXIT_FAILURE;
    }

    // Print the transformed data
    for (int i = 0; i < n_rows; ++i) {
        for (int j = 0; j < n_columns; ++j) {
            printf("(%f, %f) ", crealf(data[i * n_columns + j]), cimagf(data[i * n_columns + j]));
        }
        printf("\n");
    }

    // Clean up the GMT API session
    GMT_Destroy_Session(API);

    return EXIT_SUCCESS;
}

File: todo/test_gmt_ifft_2d.py
import numpy as np

def main():
    n_columns = 4
    n_rows = 4
    
    forward_transformed_data = np.array([
        [136 + 0j, -8 + 8j, -8 + 0j, -8 - 8j],
        [-32 + 32j, 0 + 0j, 0 + 0j, 0 + 0j],
        [-32 + 0j, 0 + 0j, 0 + 0j, 0 + 0j],
        [-32 - 32j, 0 + 0j, 0 + 0j, 0 + 0j]
    ], dtype=np.complex64)
    
    # Perform the inverse 2D FFT
    inverse_transformed_data = np.fft.ifft2(forward_transformed_data)
    
    # Print the inverse transformed data
    for i in range(n_rows):
        for j in range(n_columns):
            print("({:.1f}, {:.1f}) ".format(inverse_transformed_data[i, j].real, inverse_transformed_data[i, j].imag), end="")
        print()

if __name__ == "__main__":
    main()
